src/gui/src/layoutViewer.cpp

```diff
@@ -2547,38 +2552,61 @@ void LayoutViewer::drawBlock(QPainter* painter,
   }
   debugPrint(logger_, GUI, "draw", 1, "inst search {}", inst_timer);
 
+  utl::Timer insts_outline;
   drawInstanceOutlines(painter, insts);
+  debugPrint(logger_, GUI, "draw", 1, "inst outline render {}", insts_outline);
 
   // draw blockages
+  utl::Timer inst_blockages;
   drawBlockages(painter, block, bounds);
+  debugPrint(logger_, GUI, "draw", 1, "blockages {}", inst_blockages);
 
   dbTech* tech = block->getDataBase()->getTech();
   for (dbTechLayer* layer : tech->getLayers()) {
     drawLayer(painter, block, layer, insts, bounds, gui_painter);
   }
-  // draw instance names
+
+  utl::Timer inst_names;
   drawInstanceNames(painter, insts);
+  debugPrint(logger_, GUI, "draw", 1, "instance names {}", inst_names);
+
+  utl::Timer inst_rows;
+  drawRows(painter, block, bounds);
+  debugPrint(logger_, GUI, "draw", 1, "rows {}", inst_rows);
 
+  utl::Timer inst_access_points;
   drawRows(painter, block, bounds);
+
   if (options_->areAccessPointsVisible()) {
     drawAccessPoints(gui_painter, insts);
   }
+  debugPrint(logger_, GUI, "draw", 1, "access points {}", inst_access_points);
 
+  utl::Timer inst_module_view;
   drawModuleView(painter, insts);
+  debugPrint(logger_, GUI, "draw", 1, "module view {}", inst_module_view);
 
+  utl::Timer inst_regions;
   drawRegions(painter, block);
+  debugPrint(logger_, GUI, "draw", 1, "regions {}", inst_regions);
 
+  utl::Timer inst_pin_markers;
   if (options_->arePinMarkersVisible()) {
     drawPinMarkers(gui_painter, block, bounds);
   }
+  debugPrint(logger_, GUI, "draw", 1, "pin markers {}", inst_pin_markers);
 
+  utl::Timer inst_cell_grid;
   drawGCellGrid(painter, bounds);
+  debugPrint(logger_, GUI, "draw", 1, "save cell grid {}", inst_cell_grid);
 
+  utl::Timer inst_save_restore;
   for (auto* renderer : Gui::get()->renderers()) {
     gui_painter.saveState();
     renderer->drawObjects(gui_painter);
     gui_painter.restoreState();
   }
+  debugPrint(logger_, GUI, "draw", 1, "save restore {}", inst_save_restore);
```

Comment:
The key item is that the render is called not the save restore.  You could either rename it to 'renderers' or move the timer inside the loop and print out each renderer's time with its name.